sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -837,6 +837,48 @@ def tx_disable_channel(self, channel, disable):
 
         return self.xcvr_eeprom.write(consts.TX_DISABLE_FIELD, channel_state)
 
+    def get_rx_disable_support(self):
+        return not self.is_flat_memory() and self.xcvr_eeprom.read(consts.RX_DISABLE_SUPPORT_FIELD)
+
+    def get_rx_disable(self):
+        rx_disable_support = self.get_rx_disable_support()
+        if rx_disable_support is None:
+            return None
+        if not rx_disable_support:
+            return ["N/A" for _ in range(self.NUM_CHANNELS)]
+        rx_disable = self.xcvr_eeprom.read(consts.RX_DISABLE_FIELD)
+        if rx_disable is None:
+            return None
+        return [bool(rx_disable & (1 << i)) for i in range(self.NUM_CHANNELS)]
+
+    def rx_disable(self, rx_disable):
+        val = 0xFF if rx_disable else 0x0
+        return self.xcvr_eeprom.write(consts.RX_DISABLE_FIELD, val)
+
+    def get_rx_disable_channel(self):
+        rx_disable_support = self.get_rx_disable_support()
+        if rx_disable_support is None:
+            return None
+        if not rx_disable_support:
+            return 'N/A'
+        return self.xcvr_eeprom.read(consts.RX_DISABLE_FIELD)
+
+    def rx_disable_channel(self, channel, disable):
```

Comment:
@vdahiya12
If you make TX_DISABLE_FIELD number reg with bitfield for each channel, then you don't need to do this masking here. The bitfield will take care of this.

Use these bitfields for the tx and rx disable channel function

Reply:
https://github.com/sonic-net/sonic-platform-common/blob/94af075f83721c70ea5acd92a965d94d32421685/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L863
currently tx_disable uses a very similar approach with a mask.
so if we change this then both tx and rx should be changed similarly in my opinion.

Reply:
fixed